File: src/LeetCode/0217ContainsDuplicate.java
package LeetCode;

import java.util.*;

class ContainsDuplicate {
    public boolean containsDuplicate(int[] nums){
        Map<Integer, Integer> occurrence = new HashMap<>();
        for(int num : nums){
            if(occurrence.get(num) != null && occurrence.get(num) == 1){
                return true;
            } else{
                occurrence.put(num, 1);
            }
        }
        return false;
    }
}

File: src/LeetCode/0217ContainsDuplicate.cpp
using namespace std;

#include <vector>
#include <unordered_set>

class Solution
{
public:
    bool containsDuplicate(vector<int> &nums)
    {
        unordered_set<int> currNums;
        for (int num : nums)
        {
            if (currNums.find(num) != currNums.end())
            {
                return true;
            };

            currNums.insert(num);
        };

        return false;
    }
};